src/rust/ide/src/controller/visualisation.rs

```diff
@@ -0,0 +1,166 @@
+    , embedded_visualisers : HashMap<String,String>) -> Self {
+        Self {language_server_rpc,embedded_visualisers}
+    }
+
+    /// Get a list of all available visualisers.
+    pub async fn list_visualisers(&self) -> FallibleResult<Vec<VisualiserIdentifier>> {
+        let root_id    = self.language_server_rpc.content_root();
+        let vis_path   = language_server::Path{root_id,segments:vec!["visualisation".into()]};
+        let file_list  = self.language_server_rpc.file_list(vis_path).await?;
+        let result     = self.embedded_visualisers.keys();
+        let result     = result.map(|identifier| VisualiserIdentifier::Embedded(identifier.clone()));
+        let mut result = result.collect::<Vec<VisualiserIdentifier>>();
+        result.extend(file_list.paths.iter().filter_map(|object| {
+            if let language_server::FileSystemObject::File{name,path} = object {
+                let mut path = path.clone();
+                path.segments.push(name.to_string());
+                let root_id                  = path.root_id;
+                let segments                 = path.segments;
+                let path                     = language_server::Path{root_id,segments};
+                let visualisation_identifier = VisualiserIdentifier::File(path);
+                Some(visualisation_identifier)
+            } else {
+                None
+            }
+        }));
+        Ok(result)
+    }
+
+    /// Load the source code of the specified visualiser.
+    pub async fn load_visualiser(&self, visualiser:&VisualiserIdentifier) -> FallibleResult<String> {
+        match visualiser {
+            VisualiserIdentifier::Embedded(identifier) => {
+                let result = self.embedded_visualisers.get(identifier);
+                let error  = || {
+                    failure::Error::from(VisualiserError::NotFound{identifier:visualiser.clone()})
+                };
+                result.cloned().ok_or_else(error)
+            },
+            VisualiserIdentifier::File(path) =>
+                Ok(self.language_server_rpc.read_file(path.clone()).await?.contents)
+        }
+    }
+}
+
+
+
+// =============
+// === Tests ===
+// =============
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    use std::future::Future;
+    use utils::test::poll_future_output;
+    use enso_protocol::language_server::FileSystemObject;
+    use enso_protocol::language_server::Path;
+
+    fn result<T,F:Future<Output = FallibleResult<T>>>(fut:F) -> FallibleResult<T> {
+        let mut fut = Box::pin(fut);
+        poll_future_output(&mut fut).expect("Promise isn't ready")
+    }
+
+    #[test]
+    fn list_and_load() {
+        let mock_client     = language_server::MockClient::default();
+
+        let root_id = uuid::Uuid::default();
+        let path    = Path{root_id,segments:vec!["visualisation".into()]};
+        let paths   = vec![
+            FileSystemObject::File{path:path.clone(),name:"histogram.js".into()},
```

Comment:
could you please make a constructor, like `File::new(...)` which accepts anything that is convertible to file path? Writing `.into()` on all call-sides is an ugly API